common/lib/xmodule/xmodule/tests/__init__.py

```diff
@@ -86,6 +86,22 @@ def local_resource_url(self, block, uri):
     def get_asides(self, block):
         return []
 
+    def __repr__(self):
+        """
+        Custom hacky repr.
+        XBlock.Runtime.render() replaces the _view_name attribute while rendering, which
+        causes rendered comparisons of blocks to fail as unequal. So make the _view_name
+        attribute None during the base repr - and set it back to original value afterward.
+        """
+        orig_view_name = None
+        if hasattr(self, '_view_name'):
+            orig_view_name = self._view_name
+        self._view_name = None
+        rt_repr = super(TestModuleSystem, self).__repr__()
+        if orig_view_name:
```

Comment:
You might as well not check this condition, and always set self._view_name back to orig_view_name.